File: src/SIL.Machine.Shared/Translation/WordGraphArc.cs
using SIL.Machine.Annotations;
using System.Collections.Generic;
using System.Linq;

namespace SIL.Machine.Translation
{
	public class WordGraphArc
	{
		public WordGraphArc(int prevState, int nextState, double score, IEnumerable<string> words,
			WordAlignmentMatrix alignment, Range<int> sourceSegmentRange, bool isUnknown,
			IEnumerable<double> wordConfidences = null)
		{
			PrevState = prevState;
			NextState = nextState;
			Score = score;
			Words = words.ToArray();
			Alignment = alignment;
			SourceSegmentRange = sourceSegmentRange;
			IsUnknown = isUnknown;
			if (wordConfidences == null)
				WordConfidences = Enumerable.Repeat(-1.0, Words.Count).ToList();
			else
				WordConfidences = wordConfidences.ToList();
		}

		public int PrevState { get; }
		public int NextState { get; }
		public double Score { get; }
		public IReadOnlyList<string> Words { get; }
		public WordAlignmentMatrix Alignment { get; }
		public IList<double> WordConfidences { get; }
		public Range<int> SourceSegmentRange { get; }
		public bool IsUnknown { get; }
	}
}

File: src/SIL.Machine.TS/src/translation/word-graph-arc.ts
import { Range } from '../annotations/range';
import { WordAlignmentMatrix } from './word-alignment-matrix';

export class WordGraphArc {
  constructor(
    public readonly prevState: number,
    public readonly nextState: number,
    public readonly score: number,
    public readonly words: string[],
    public readonly alignment: WordAlignmentMatrix,
    public readonly sourceSegmentRange: Range,
    public readonly isUnknown: boolean,
    public readonly wordConfidences: number[] = new Array<number>(words.length).fill(-1)
  ) {}
}
